drivers/serial/uart_nrf5.c

```diff
@@ -221,29 +162,34 @@ static int uart_nrf5_init(struct device *dev)
 				  CONFIG_UART_NRF5_GPIO_CTS_PIN,
 				  (GPIO_DIR_IN));
 
-	uart->PSELRTS = CONFIG_UART_NRF5_GPIO_RTS_PIN;
-	uart->PSELCTS = CONFIG_UART_NRF5_GPIO_CTS_PIN;
-	uart->CONFIG = (UART_CONFIG_HWFC_Enabled << UART_CONFIG_HWFC_Pos);
-
+	nrf_uart_hwfc_pins_set(uart,
+			       CONFIG_UART_NRF5_GPIO_RTS_PIN,
+			       CONFIG_UART_NRF5_GPIO_CTS_PIN);
+	nrf_uart_configure(uart,
+			   NRF_UART_PARITY_EXCLUDED,
+			   NRF_UART_HWFC_ENABLED);
+#else
+	nrf_uart_configure(uart,
+			   NRF_UART_PARITY_EXCLUDED,
+			   NRF_UART_HWFC_DISABLED);
 #endif /* CONFIG_UART_NRF5_FLOW_CONTROL */
 
 	DEV_DATA(dev)->baud_rate = CONFIG_UART_NRF5_BAUD_RATE;
 
 	/* Set baud rate */
-	err = baudrate_set(dev, DEV_DATA(dev)->baud_rate,
-		     DEV_CFG(dev)->sys_clk_freq);
+	err = baudrate_set(dev, DEV_DATA(dev)->baud_rate);
```

Comment:
It seems pointless to dedicate the `baud_rate` field for this. You can use `CONFIG_UART_NRF5_BAUD_RATE` directly here.